src/main/kotlin/com/asap/asapbackend/domain/announcement/presentation/SchoolAnnouncementController.kt

```diff
@@ -0,0 +1,94 @@
+package com.asap.asapbackend.domain.announcement.presentation
+
+import com.asap.asapbackend.domain.announcement.application.SchoolAnnouncementService
+import com.asap.asapbackend.domain.announcement.application.dto.*
+import io.github.oshai.kotlinlogging.KotlinLogging
+import org.springframework.web.bind.annotation.*
+
+private val logger = KotlinLogging.logger {}
+
+@RestController
+class SchoolAnnouncementController(
+    private val announcementService: SchoolAnnouncementService
+) {
+
+
+    @GetMapping(AnnouncementApi.V1.SCHOOL_ANNOUNCEMENT)
+    fun getSchoolAnnouncement(
+        @RequestParam size: Int,
+        @RequestParam page: Int
+    ): GetSchoolAnnouncementSlice.Response {
+        return announcementService.getSchoolAnnouncementSlice(GetSchoolAnnouncementSlice.Request(size, page))
+    }
+
+    @GetMapping(AnnouncementApi.V1.SCHOOL_ANNOUNCEMENT_DETAIL)
+    fun getSchoolAnnouncementDetail(
+        @PathVariable schoolAnnouncementId: Long
+    ): GetSchoolAnnouncementDetail.Response {
+        return announcementService.getSchoolAnnouncementDetail(GetSchoolAnnouncementDetail.Request(schoolAnnouncementId))
+    }
+
+
+    @GetMapping(AnnouncementApi.V1.EDUCATION_OFFICE_ANNOUNCEMENT)
+    fun getEducationOfficeAnnouncement(
+        @RequestParam size: Int,
+        @RequestParam page: Int
+    ): GetEducationOfficeAnnouncementSlice.Response {
+        return announcementService.getEducationOfficeAnnouncementSlice(
+            GetEducationOfficeAnnouncementSlice.Request(
+                size,
+                page
+            )
+        )
+    }
+
+
+    @GetMapping(AnnouncementApi.V1.EDUCATION_OFFICE_ANNOUNCEMENT_DETAIL)
+    fun getEducationOfficeAnnouncementDetail(
+        @PathVariable educationOfficeAnnouncementId: Long
+    ): GetEducationOfficeAnnouncementDetail.Response {
+        return announcementService.getEducationOfficeAnnouncementDetail(
+            GetEducationOfficeAnnouncementDetail.Request(
+                educationOfficeAnnouncementId
+            )
+        )
+    }
+
+
+    @PatchMapping(AnnouncementApi.V1.SCHOOL_ANNOUNCEMENT_CATEGORY)
+    fun patchSchoolAnnouncementCategory(
+        @PathVariable schoolAnnouncementId: Long,
+        @RequestBody updateRequest: UpdateSchoolAnnouncementCategory.Request
```

Comment:
얘는 왜 Request안에 넣어서 requestBody로 받는거야?
```suggestion
        @RequestParam category: AnnouncementCategory
```
파라미터로 받는게 좋지 않을까? 혹시 이것도 id를 path에 넣는것처럼 기준이 있었어?

Reply:
값이 변경되는데 그 값을 파라미터에 넣는거는 매우매우매우매우매우 안좋은것같아

Reply:
오히려 저렇게하면 변경 정보가 유출될 수 있어